safenode/src/node/api.rs

```diff
@@ -160,44 +122,207 @@ impl Node {
                 self.send_response(Response::Query(resp), response_channel)
                     .await;
             }
+            Request::Event(event) => {
+                match event {
+                    Event::DoubleSpendAttempted(a_spend, b_spend) => {
+                        let a_addr = dbc_address(a_spend.dbc_id());
+                        let b_addr = dbc_address(b_spend.dbc_id());
+
+                        if a_addr == b_addr {
+                            // We carelessly add the two spends here as they will automatically be
+                            // marked as double spend attempt.
+                            self.try_add_double_if_different_hash(&a_spend, &b_spend)
+                                .await?;
+                        } else {
```

Comment:
Why do we handle this? Can't we just ignore it since the format is unexpected? Instead of trying to fix it and bcasting it to peers?

Reply:
Yeah why not?

If we find a double spend then of course we shall broadcast it.

Reply:
Our code won't send anything else. 
Even if restricting how it's instantiated it doesn't change that our code won't send anything else.

So, it's merely to handle that case. We can drop those as well.

Reply:
Updated to drop those now.